src/dxgi/dxgi_adapter.cpp

```diff
@@ -145,14 +145,71 @@ namespace dxvk {
     
     if (ppOutput == nullptr)
       return E_INVALIDARG;
-    
-    HMONITOR monitor = wsi::enumMonitors(Output);
-    
-    if (monitor == nullptr)
+
+    const auto &deviceId   = m_adapter->devicePropertiesExt().vk11;
+
+    if (!deviceId.deviceLUIDValid)
       return DXGI_ERROR_NOT_FOUND;
-    
-    *ppOutput = ref(new DxgiOutput(m_factory, this, monitor));
-    return S_OK;
+
+    std::vector<DISPLAYCONFIG_PATH_INFO> paths;
+    std::vector<DISPLAYCONFIG_MODE_INFO> modes;
+    UINT32 pathCount,modeCount;
+    LONG result;
+
+    do
+    {
+        if ((result = GetDisplayConfigBufferSizes(QDC_ONLY_ACTIVE_PATHS, &pathCount, &modeCount)))
+        {
+          Logger::err(str::format("GetDisplayConfigBufferSizes failed, result ", result));
+          return DXGI_ERROR_NOT_FOUND;
+        }
+        paths.resize(pathCount);
+        modes.resize(modeCount);
+        result = QueryDisplayConfig(QDC_ONLY_ACTIVE_PATHS, &pathCount, paths.data(), &modeCount, modes.data(), nullptr);
+    } while (result == ERROR_INSUFFICIENT_BUFFER);
+
+    if (result)
+    {
+      Logger::err(str::format("QueryDisplayConfig failed, result ", result));
+      return DXGI_ERROR_NOT_FOUND;
+    }
+    paths.resize(pathCount);
+    modes.resize(modeCount);
+
+    MonitorEnumInfo info;
+    info.iMonitorId = Output;
+    info.currId = 0;
+    info.oMonitor   = nullptr;
+
+    for (const auto &path : paths)
+    {
+      if (std::memcmp(&path.sourceInfo.adapterId, deviceId.deviceLUID, VK_LUID_SIZE))
+        continue;
+
+      DISPLAYCONFIG_SOURCE_DEVICE_NAME deviceName;
+
+      deviceName.header.adapterId = path.sourceInfo.adapterId;
+      deviceName.header.id = path.sourceInfo.id;
+      deviceName.header.type = DISPLAYCONFIG_DEVICE_INFO_GET_SOURCE_NAME;
+      deviceName.header.size = sizeof(deviceName);
+      if ((result = DisplayConfigGetDeviceInfo(&deviceName.header)))
+      {
+        Logger::err(str::format("DisplayConfigGetDeviceInfo failed, result ", result));
+        return DXGI_ERROR_NOT_FOUND;
+      }
+      info.gdiDeviceName = deviceName.viewGdiDeviceName;
```

Comment:
this seems to be unused by MonitorEnumProc, is this on purpose?

Reply:
Oh, that is the consequences of too formal / quick rebase... The original patches (ad38417134fccaa2a8a377225151ed529681f324, 4fc55c2f90b0b560d8f5e33cbab3bdb72c249686) of course filter by this gdiDeviceName  in enum function:
```
    monitorInfo.cbSize = sizeof(monitorInfo);
    GetMonitorInfoW(hmon, (MONITORINFO *)&monitorInfo);
    if (wcscmp(data->gdiDeviceName, monitorInfo.szDevice))
      return TRUE;
```

I should probably move this part to wsi:: anyway, passing dxgi adapter luid to wsi::enumMonitors()?

Reply:
> passing dxgi adapter luid to wsi::enumMonitors()

Sounds good, this should solve the portability issues.

Reply:
Yeah, makes sense. Except for if can't find any GDI device in the QueryDisplayConfig, we have to report zero output count for this thing to work. QueryDisplayConfig queries only active display paths and if some adapter has no outputs it won't be returned there. It is possible to query more on Windows (inactive paths) but it is currently not supported under Wine and we are actually interested in active outputs only here.